core/agents/bug_hunter.py

```diff
@@ -0,0 +1,190 @@
+                docs=self.current_state.docs,
+                next_solution_to_try=None,
+            )
+        )
+        bug_reproduction_instructions = await llm(convo, temperature=0)
+        self.next_state.current_iteration["bug_reproduction_description"] = bug_reproduction_instructions
+
+    async def check_logs(self, logs_message: str = None):
+        llm = self.get_llm()
+        convo = (
+            AgentConvo(self)
+            .template(
+                "iteration",
+                current_task=self.current_state.current_task,
+                user_feedback=self.current_state.current_iteration["user_feedback"],
+                user_feedback_qa=self.current_state.current_iteration["user_feedback_qa"],
+                docs=self.current_state.docs,
+                magic_words=magic_words,
+                next_solution_to_try=None
+            )
+        )
+
+        for hunting_cycle in self.current_state.current_iteration["bug_hunting_cycles"]:
+            convo = (convo
+                     .assistant(hunting_cycle["human_readable_instructions"])
+                     .template(
+                        "log_data",
+                        backend_logs=hunting_cycle["backend_logs"],
+                        frontend_logs=hunting_cycle["frontend_logs"],
+                        fix_attempted=hunting_cycle["fix_attempted"]
+                    ))
+
+        human_readable_instructions = await llm(convo, temperature=0.5)
+
+        convo = (
+            AgentConvo(self)
+            .template(
+                "bug_found_or_add_logs",
+                hunt_conclusion=human_readable_instructions,
+            )
+            .require_schema(HuntConclusionOptions)
+        )
+        hunt_conclusion = await llm(convo, parser=JSONParser(HuntConclusionOptions), temperature=0)
+
+        self.next_state.current_iteration["description"] = human_readable_instructions
+        self.next_state.current_iteration["bug_hunting_cycles"] += [{
+            "human_readable_instructions": human_readable_instructions,
+            "fix_attempted": False
+        }]
+
+        if False and hunt_conclusion.conclusion == magic_words.PROBLEM_IDENTIFIED:
+            # if no need for logs, implement iteration same as before
+            self.next_state.current_iteration["status"] = IterationStatus.AWAITING_BUG_FIX
+            await self.send_message("The bug is found - I'm  attempting to fix it.")
+        else:
+            # if logs are needed, add logging steps
+            self.next_state.current_iteration["status"] = IterationStatus.AWAITING_LOGGING
+            await self.send_message("Adding more logs to identify the bug.")
+
+        self.next_state.flag_iterations_as_modified()
+        return AgentResponse.done(self)
+
+    async def ask_user_to_test(self):
+
+        reproduce_bug_and_get_logs = self.current_state.current_iteration["status"] == IterationStatus.AWAITING_BUG_REPRODUCTION
```

Comment:
This is basically variable that i mentioned to pass in to the function so we dont check status here

Reply:
yup